ImGuiExtension.cs

```
@@ -1,129 +1,125 @@
-        public static DisposableImGui UColumns( int count, string id = null, bool border = false )
-        {
-            Columns( count, id, border );
-            return new DisposableImGui( true, DisposableTypes.Columns );
-        }
-        public static DisposableImGui Window( string name, ref bool open, out bool collapsed, ImGuiWindowFlags flags = ImGuiWindowFlags.None )
+        PushID(id);
+        return new DisposableImGui(true, DisposableTypes.ID);
+    }
+    public static DisposableImGui ID(int id)
+    {
+        PushID(id);
+        return new DisposableImGui(true, DisposableTypes.ID);
+    }
+    public static DisposableImGui UCombo(string label, string previewValue, out bool open, ImGuiComboFlags flags = ImGuiComboFlags.None)
+    {
+        return new DisposableImGui(open = BeginCombo(label, previewValue, flags), DisposableTypes.Combo);
+    }
+    public static DisposableImGui Tooltip()
+    {
+        BeginTooltip();
+        return new DisposableImGui(true, DisposableTypes.Tooltip);
+    }
+    public static DisposableImGuiIndent UIndent(float size = 0f) => new DisposableImGuiIndent(size);
+    public static DisposableImGui UColumns(int count, string id = null, bool border = false)
+    {
+        Columns(count, id, border);
+        return new DisposableImGui(true, DisposableTypes.Columns);
+    }
+    public static DisposableImGui Window(string name, ref bool open, out bool collapsed, ImGuiWindowFlags flags = ImGuiWindowFlags.None)
+    {
+        collapsed = !Begin(name, ref open, flags);
+        return new DisposableImGui(true, DisposableTypes.Window);
+    }
+
+    public static unsafe DisposableImGui Child(string cln = "", Vector2 size = default,
+        ImGuiChildFlags childFlags = ImGuiChildFlags.None, ImGuiWindowFlags flags = ImGuiWindowFlags.None)
+    {
+        BeginChild(cln, size, childFlags, flags);
+        return new DisposableImGui(true, DisposableTypes.Child);
+    }
```

The `CallerLineNumber` here transparently assigns distinct ids for each call to `BeginChild`, having distinct values is important for imgui to process inputs/events from what I remember. We can't keep this as is since it will now always create conflicts by default and hides away the best practice version of the call.

---

ah my mistake, this was one of the first things I changed before I had a better understanding. Should be back to before now apart from the add childflags property.